### DISCUSSION TITLE
How do I read data from a BNO055 Xplained Pro board using a TTL-232RG-VREG3V3-WE  cable and YARP?

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#112] [Q&A] [answered]
See title.

**Comment by traversaro** [comment, marked as answer]
Datasheet BNO055 : https://www.bosch-sensortec.com/bst/products/all_products/bno055 .
Docs Atmel BNO055 Xplained Pro : http://www.atmel.com/tools/ATBNO055-XPRO.aspx?tab=documents
TTL-232RG-VREG3V3-WE  cable : http://www.ftdichip.com/Products/Cables/USBTTLSerial.htm

# Connect the  TTL-232RG-VREG3V3-WE   to the BNO055 Xplained Pro pinout

| TTL-232RG-VREG3V3-WE | BNO055 Xplained Pro |
| :-: | :-: |
| RED | 20 |
| BLACK | 2 |
| YELLOW | 11 |
| ORANGE | 12 |

![bno055-ttl-232r-3v3](https://cloud.githubusercontent.com/assets/1857049/15151381/28f7dff8-16d1-11e6-9825-c153569bb351.jpg)

# Prepare YARP

Compile YARP with the `imuBosch_BNO055` device, setting to `ON` the `ENABLE_yarpmod_imuBosch_BNO055` CMake option.

# Launch the yarpserver (if it is not already running)

```
yarpserver
```

# Launch the driver

```
yarpdev --device inertial --subdevice imuBosch_BNO055 --comport /dev/ttyUSB0 --channels 16 --name /seesaw
```

# Read the data

```
yarp read ... /seesaw 
```

cc @DanielePucci @francesco-romano @nunoguedelha and kudos to @barbalberto @maggia80

**Comment by maggia80** [comment]
actually the FTDI cable you are using is not correct since the Vcc is 5V, only the TX and RX has the 3V3 levels.

The right cable is the TTL-232RG-VREG3V3-WE

**Comment by traversaro** [comment]
I check the document and the pinout should be consistent, however this still need to be tested.
